### PR TITLE
fix data race on makeCapacity & fakeController

### PR DESCRIPTION
**Pull request by dfajmon**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
When running tests with `-race` go routine leaks on `capacityCounter` & `utilruntime.ReallyCrash`
Creating mutex for these variables fixes it.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/capacity/capacity_test.go [hunk 1]
```diff
@@ -1608,9 +1612,13 @@ func str2quantity(str string) *resource.Quantity {
 	return &quantity
 }
 
+var counterMutex sync.Mutex
 var capacityCounter int
 
 func makeCapacity(in testCapacity) *storagev1.CSIStorageCapacity {
+	counterMutex.Lock()
+	defer counterMutex.Unlock()
+	// This is a global counter, that goroutines access causing data race
```
Comment (gnufied) [inline pkg/capacity/capacity_test.go:1621]: How about something like https://gist.github.com/gnufied/f87cb7c574fc74b7b16797fd2d95ac71 ?

Comment (dfajmon) [reply pkg/capacity/capacity_test.go]: updated, thanks for help

File: pkg/capacity/capacity_test.go [hunk 2]
```diff
@@ -1608,10 +1605,28 @@ func str2quantity(str string) *resource.Quantity {
 	return &quantity
 }
 
-var capacityCounter int
+type capacityCount struct {
+	sync.RWMutex
+	capacityCounter int
+}
```
[the remark this answers is not in the document]
Comment (dfajmon) [reply pkg/capacity/capacity_test.go]: sure, updated

**Pull request merged**